src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -8256,31 +8256,37 @@ export class StudyViewPageStore
         default: {},
     });
 
-    readonly getDataForClinicalDataTab = remoteData({
-        await: () => [this.clinicalAttributes, this.selectedSamples],
+    readonly getDataForClinicalDataTab: MobxPromiseUnionTypeWithDefault<
+        { [attr: string]: string }[]
+    > = remoteData({
+        await: () => [
+            this.clinicalAttributes,
+            this.selectedSamples,
+            this.sampleSetByKey,
+        ],
         onError: () => {},
         invoke: async () => {
             if (this.selectedSamples.result.length === 0) {
                 return Promise.resolve([]);
             }
             let sampleClinicalDataMap: {
-                [attributeId: string]: { [attributeId: string]: string };
-            } = await getClinicalDataBySamples(this.selectedSamples.result);
-            return _.reduce(
-                this.selectedSamples.result,
-                (acc, next) => {
-                    let sampleData: { [attributeId: string]: string } = {
-                        studyId: next.studyId,
-                        patientId: next.patientId,
-                        sampleId: next.sampleId,
-                        ...(sampleClinicalDataMap[next.uniqueSampleKey] || {}),
-                    };
+                [sampleId: string]: { [attributeId: string]: string };
```

Comment:
again, this type should be inferred by typescript from internals or definition of function ```getAllClinicalDataByStudyViewFilter```

Reply:
Ok, I removed this type declaration.